### PR TITLE
Rewrite codeSonarData

### PR DESCRIPTION
**Pull request by doudou**

### CODE REVIEW

**Comment by doudou** [comment]
don't forget to push the bugfixes to codeSonarData.

**Comment by romulogcerqueira** [comment]
Sure, @doudou. I will update soon.

**Comment by romulogcerqueira** [comment]
Updated!

File: src/MultibeamSonar.cpp [hunk 1]
```diff
@@ -28,52 +28,26 @@ base::samples::Sonar MultibeamSonar::simulateMultiBeam(const std::vector<float>&
...
-	}
-
-	return sonar_data;
+std::vector<float> MultibeamSonar::codeSonarData(const cv::Mat3f& cv_image)
+{
+    // associates shader columns with their respective beam
+    std::vector<float> sonar_data(_beam_count * _bin_count, 0.0);
+
+    double const beam_size = _beam_width.rad / _beam_count;
+    double const half_fovx = beam_size * _beam_count / 2;
```
Comment (doudou) [inline src/MultibeamSonar.cpp]: This is `_beam_width.getRad() / 2;`

File: src/MultibeamSonar.cpp [hunk 2]
```diff
@@ -28,52 +28,26 @@ base::samples::Sonar MultibeamSonar::simulateMultiBeam(const std::vector<float>&
...
-		}
-	}
-
-	return sonar_data;
+std::vector<float> MultibeamSonar::codeSonarData(const cv::Mat3f& cv_image)
+{
+    // associates shader columns with their respective beam
+    std::vector<float> sonar_data(_beam_count * _bin_count, 0.0);
+
+    double const beam_size = _beam_width.rad / _beam_count;
```
Comment (doudou) [inline src/MultibeamSonar.cpp]: use `getRad()` instead of accessing `rad` directly.

Comment (romulogcerqueira) [inline src/MultibeamSonar.cpp]: OK. Done!

File: src/MultibeamSonar.cpp (hunk 1 above)
Comment (romulogcerqueira) [inline src/MultibeamSonar.cpp]: It will present the same result, but it is more readable. Done!

**Pull request merged**